src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/Type.cs

```diff
@@ -404,7 +366,7 @@ public ConstValKind constValKind()
         }
         public CType underlyingType()
         {
-            if (IsAggregateType() && getAggregate().IsEnum())
+            if (this is AggregateType && getAggregate().IsEnum())
```

Comment:
#16664 covers getting away from artefacts of the assembly's having been ported from another language and gradually being more idiomatic C#. I'm not sure if everyone would agree that favouring virtual methods over test and dispatch counts, but it does to my mind. That issue is a bit open-ended and subjective though.